sketch/README.md

```diff
@@ -0,0 +1,197 @@
+There are allot of ‘AT’ commands you can send to the device, to configure or check settings. A good reference guide can be found here: [AT commands guide](https://room-15.github.io/blog/2015/03/26/esp8266-at-command-reference/)
+
+A good response looks like this.
+
+<img src="resources/ATOK.png" alt="Image of a good response" width="1000" align="middle">
+
+If you are only getting back garbage values, like seen here.
+
+<img src="resources/WrongBaudrate.png" alt="Image of a garbage response" width="1000" align="middle">
+
+Some extra hardware will be needed. For our use case, a FTDI programmer was used. There are other methods to configure the ESP. Like comunicate with it through an [*arduino*](http://www.martyncurrey.com/arduino-to-esp8266-serial-commincation/), but we will only be looking at the FTDI one. 
+
+*NOTE: It is worth mentioning that if the baudrate is indeed 115200 the example used with the arduino is not going to work, as it is using the softserial library, which can only go up to 57600. 
+A work around would be to use a mega which has multiple hardware serials, which can handle the 115200 baudrate.*
+
+With the FTDI it is possible to connect the ESP directly to a pc, and issue commands.
+
+Here is how you connect the FTDI:
+ 
+<img src="resources/FDTI-to-ESP.png" alt="FTDI hookup" width="700" align="middle">
+ 
+When the ESP is connected to your pc through your FTDI, it is possible to communicate with the device. This can be done with your preferred serial terminal program, *E.G.* Putty, Termite, Minicom. To name a few. 
+It is also possible to use the arduino serial monitor. The only parameter you need to know is the serial port with which the ESP is connected and the baud rate. 
+The serial port can be found in the Arduino IDE, under the `tools section`,
+
+<img src="resources/ChoosePort.png" alt="Image of Arduino choose port" width="300" align="middle">
+
+or in the `device manager` (Windows)
+
+<img src="resources/DeviceManager.png" alt="Image of device manager" width="300" align="middle">
+
+On Mac the easiest way is to open up the terminal and type in the following command: ```ls /dev/cu.*```.
+this should list the ports available.
+
+The different baud rates can be tried until one that works is found.
+
+### Important values/calculations
+---
+`#define HEX_CHAR_TO_NIBBLE(c) ((c >= 'a') ? (c - 'a' + 0x0A) : (c - '0'))`
+
+`#define HEX_PAIR_TO_BYTE(h, l) ((HEX_CHAR_TO_NIBBLE(h) << 4) + HEX_CHAR_TO_NIBBLE(l))`
+
+The above calculations turn a hex value into a byte value. In the `HEX_PAIR_TO_BYTE` the `HEX_CHAR_TO_NIBBLE` is called. Here a single char is evaluated and changed accordingly.
+
+`#define ESP_SERIAL_SPEED 9600`
+
+The standard baudrate of most ESP8266-01 module is 115200. This can be permanently changed by communicating to the device directly and using the following command: **AT+UART_DEF=9600,8,1,0,0** (Note: The default speed cannot be used with the explorer board, as its serial port can only handle up to 57600).
```

Comment:
Isn't this duplicate with what's stated above?

Reply:
Moved it up to the communication part and rephrased